src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopyArray = [];
+  const stateCopy = Object.assign({}, state);
+
+  for (const i of actions) {
+    switch (i.type) {
+      case 'addProperties': Object.assign(stateCopy, i.extraData);
+        break;
```

Comment:
let's move to the new line, more readable
```suggestion
      case 'addProperties': 
        Object.assign(stateCopy, i.extraData);
        break;
```